### PR TITLE
fix(site): make tag pills horizontally scrollable

### PR DESCRIPTION
**Pull request by dante01yoon**
## Summary
- Fix tag pills on workflow cards being clipped by `overflow-hidden` on narrow screens
- Change to `overflow-x-auto` with hidden scrollbar so users can swipe/scroll to see all tags
- Apply `shrink-0 whitespace-nowrap` to tag badges to prevent shrinking and text truncation
- Add E2E tests verifying tag container overflow property and flex-shrink behavior

## Test plan
- [x] On mobile, swipe horizontally on cards with 3+ tags
- [x] On desktop, scroll tags with trackpad/mouse wheel
- [x] Verify scrollbar is not visible
- [x] E2E tests pass: `pnpm run test:e2e`

## demo

https://github.com/user-attachments/assets/44ca79a4-ed97-475e-8aaa-adcb63f17cdb

Slack ref: https://comfy-organization.slack.com/archives/C0AEPRS8N74/p1773076747344099

notion ref: https://www.notion.so/comfy-org/Bug-Filter-tags-cut-off-on-workflow-cards-31e6d73d3650815e9a09eeffee7bd1b5?source=copy_link

### CODE REVIEW

**Review by Myestery — CHANGES_REQUESTED** [review]

File: site/tests/e2e.spec.ts [hunk 1]
```diff
@@ -216,6 +216,41 @@ test.describe('Search Filter "+X more" Expansion', () => {
...
+  test('tag container allows horizontal scroll instead of clipping', async ({ page }) => {
+    await page.goto('/workflows/');
+    const firstCard = templateCardLink(page);
+    await expect(firstCard).toBeAttached({ timeout: 10000 });
+
+    // Find a tag container inside a card
+    const tagContainer = page.locator('[data-testid="tag-pills"]').first();
+    await expect(tagContainer).toBeAttached({ timeout: 5000 });
+
+    const overflowX = await tagContainer.evaluate(
```
[the remark this answers is not in the document]
Comment (dante01yoon) [reply site/tests/e2e.spec.ts]: Addressed

**Review by Myestery — APPROVED** [review]
The last comment is non blocking and can be done in a follow up PR

File: site/tests/e2e.spec.ts [hunk 2]
```diff
@@ -216,6 +216,51 @@ test.describe('Search Filter "+X more" Expansion', () => {
...
+    await lastTag.evaluate((el) =>
+      el.scrollIntoView({ behavior: 'instant', block: 'nearest', inline: 'nearest' })
+    );
+    await Promise.all([
+      page.waitForNavigation(),
+      lastTag.click(),
+    ]);
+
+    // Assert navigation went to the tag filter page
+    expect(page.url()).toMatch(/\/workflows\/tag\/.+\//);
```
[the remark this answers is not in the document]
Comment (dante01yoon) [reply site/tests/e2e.spec.ts]: @coderabbitai create issue for follow up
